core/tauri-codegen/src/context.rs

```diff
@@ -289,17 +282,9 @@ pub fn context_codegen(data: ContextData) -> Result<TokenStream, EmbeddedAssetsE
   let with_tray_icon_code = if target.is_desktop() {
     if let Some(tray) = &config.app.tray_icon {
       let tray_icon_icon_path = config_parent.join(&tray.icon_path);
-      let ext = tray_icon_icon_path.extension();
-      if ext.map_or(false, |e| e == "ico") {
-        ico_icon(&root, &out_dir, tray_icon_icon_path)
-          .map(|i| quote!(context.set_tray_icon(Some(#i));))?
-      } else if ext.map_or(false, |e| e == "png") {
-        png_icon(&root, &out_dir, tray_icon_icon_path)
-          .map(|i| quote!(context.set_tray_icon(Some(#i));))?
-      } else {
-        quote!(compile_error!(
-          "The tray icon extension must be either `.ico` or `.png`."
-        ))
+      match image_icon(&root, &out_dir, &tray_icon_icon_path) {
+        Ok(i) => quote!(context.set_tray_icon(Some(#i));),
+        Err(e) => return Err(e),
       }
```

Comment:
we need to add back the compile_err for the icon extension or if the error variant already contains this message we can just do

```suggestion
      image_icon(&root, &out_dir, &tray_icon_icon_path).map(|i| quote!(context.set_tray_icon(Some(#i));))?
```